lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Monitoring/Http/SendFaucetAssets.hs

```diff
@@ -0,0 +1,256 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Wallet.Launch.Cluster.Monitoring.Http.SendFaucetAssets
+    ( SendFaucetAssets (..)
+    , WithNetwork (..)
+    , genSendFaucetAssets
+    )
+where
+
+import Prelude
+
+import Cardano.Wallet.Address.Encoding
+    ( decodeAddress
+    , encodeAddress
+    )
+import Cardano.Wallet.Faucet.Gen.Address
+    ( NetworkTag (..)
+    , genAddress
+    )
+import Cardano.Wallet.Launch.Cluster.Monitoring.Http.OpenApi
+    ( sendAssetsSchema
+    )
+import Cardano.Wallet.Primitive.NetworkId
+    ( HasSNetworkId (sNetworkId)
+    , NetworkDiscriminant (..)
+    , SNetworkId (..)
+    )
+import Cardano.Wallet.Primitive.Types.Address
+    ( Address (..)
+    )
+import Cardano.Wallet.Primitive.Types.AssetId
+    ( AssetId (..)
+    )
+import Cardano.Wallet.Primitive.Types.Coin
+    ( Coin (Coin)
+    )
+import Cardano.Wallet.Primitive.Types.TokenBundle
+    ( TokenBundle
+    , fromFlatList
+    , toFlatList
+    )
+import Cardano.Wallet.Primitive.Types.TokenBundle.Gen
+    ( genTokenBundleSmallRange
+    )
+import Cardano.Wallet.Primitive.Types.TokenQuantity
+    ( TokenQuantity
+    )
+import Cardano.Wallet.Util
+    ( ShowFmt (..)
+    )
+import Control.Monad
+    ( (>=>)
+    )
+import Data.Aeson
+    ( FromJSON (parseJSON)
+    , KeyValue ((.=))
+    , ToJSON (toJSON)
+    , Value
+    , object
+    , withObject
+    , (.:)
+    )
+import Data.Aeson.Types
+    ( Parser
+    )
+import Data.Bifunctor
+    ( first
+    )
+import Data.OpenApi
+    ( NamedSchema (..)
+    , ToSchema (..)
+    )
+import Data.Typeable
+    ( Typeable
+    )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Gen
+    , listOf
+    )
+
+import qualified Cardano.Address as Addr
+
+-- | Payload to send assets to a list of addresses
+data SendFaucetAssets = SendFaucetAssets
+    { batchSize :: Int
+    -- ^ batch size
+    , assets :: [(Address, (TokenBundle, [(String, String)]))]
+    -- ^ List of addresses and the assets to send to each address
+    }
+    deriving stock (Eq, Show)
```

Comment:
```suggestion
type AssetMetadata = [(String, String)]

-- | Payload to send assets to a list of addresses
data SendFaucetAssets = SendFaucetAssets
    { batchSize :: Int
    -- ^ batch size
    , assets :: [(Address, (TokenBundle, AssetMetadata))]
    -- ^ List of addresses and the assets to send to each address
    }
    deriving stock (Eq, Show)
```

Type synonym for documentation purposes, makes it easier to understand what this list is supposed to be.